implement the cdn-fc-plugin

Fitz-devs opened the pull request:
first pr

Commit:
feat: implement cdn-fc-plugin

Commit:
other: add example

Commit:
bugfix: fix the wrong Category

heimanba committed:
add plugin

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+const core = require("@serverless-devs/core");
+const { CatchableError, lodash, fse, rimraf, getCredential, spinner, Logger } = core;
```
const core = require("@serverless-devs/core");
const { CatchableError, lodash, fse, rimraf, getCredential, spinner, Logger } = core; 

可以写成一行哈

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+const logger = new Logger("cdn-fc-plugin");
+const  fc = require('@alicloud/fc-open20210406')
+const FC_Open20210406 = fc.default;
+const ListCustomDomainsRequest = fc.ListCustomDomainsRequest;
```
4、5、6 能直接解构出来吗？

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+const FC_Open20210406 = fc.default;
+const ListCustomDomainsRequest = fc.ListCustomDomainsRequest;
+const Config = require('@alicloud/openapi-client').Config;
+const SPINNER_VM = spinner('start action')
```
这个建议写在函数里面哈，如果说别人引用了你这个文件，还没有开始执行呢 这个 spinner 就开始运行了

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+
+module.exports = async function index(inputs, args) {
+    const serviceName = args.serviceName;
+    const functionName = args.functionName;
```
建议结构赋值

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+        securityToken: credentials.SecurityToken,
+    });
+    // 访问的域名
+    config.endpoint = `1656564022886445.cn-shenzhen.fc.aliyuncs.com`;
```
这个是写死的？

wss-git replied:
应该是  `${accoundId}.${region}.fc.aliyuncs.com`

Fitz-devs replied:
这个是写死的，这应该是openapi的地址

wss-git replied:
不是的哈，这个是 uid + region + 固定的后缀

wss-git commented on src/index.js:
```diff
@@ -0,0 +1,71 @@
+    config.endpoint = `1656564022886445.cn-shenzhen.fc.aliyuncs.com`;
+
+    const client = new FC_Open20210406(config);
+    let listCustomDomainsRequest = new ListCustomDomainsRequest({ prefix: `${functionName}.${serviceName}`,  limit: 1});
```
这个有点没有看明白。
不过我理解你是想找 函数和服务名称开头的？  如果这样这个不合理哈，因为大小写的原因和自定义的配置 可能会导致最终的函数对应不上。
建议交互：
方式1.  直接支持传入 custom domain
方式2.  list domains，然后过滤 rutes 是否包含和指定的函数（最好细致到指定的版本和别名）

Fitz-devs replied:
对于方法1，如果传入custom domain了，就不需要这个插件了；
方法2，所有的 list domain的确是可行的，但我觉得既然服务名和函数名是不可变的，而生成的域名又是有规律的，感觉没有必要。

wss-git replied:
有用户会拿着生成的域名重新绑定到其他的函数上哈，这样cdn 就对映不上了 @heimanba 麻烦看看

Commit:
bugfix: fix the problem was mentioned in pr1

Commit:
Merge remote-tracking branch 'origin/develop' into develop

Commit:
bugfix: fix the `Cannot find module xxx` bug

Commit:
feat: 针对有多个自定义域名的服务名和函数匹配的情况提出解决方案
+ 自定义域名的路由规则中存在一条路由的服务名，函数名与目标服务名和函数名相同且路由路径为`/*`直接设置为主源
+ 当自定义域名的路由规则的所有服务名都为目标服务名时，也可作为源，默认为备源，如所有源都为备源时，最新的一个源会改为主源
+ 权重按自定义域名由新到旧排序，新的自定义域名权重越大
